src/libs/Navigation/NavigationRoot.js

```diff
@@ -50,14 +55,24 @@ function parseAndLogRoute(state) {
 
 const NavigationRoot = (props) => {
     useFlipper(navigationRef);
+    const navigationStateRef = useRef(undefined);
+
+    const updateSavedNavigationStateAndLogRoute = (state) => {
+        navigationStateRef.current = state;
+        props.updateCurrentReportId(state);
+        parseAndLogRoute(state);
+    };
+
     return (
         <NavigationContainer
+            key={props.isSmallScreenWidth ? 'small' : 'big'}
```

Comment:
This causes tons of issues when resize browser. I know this was intentional but is there alternative solution removing this line? (tagging all navigation reboot experts)
- https://github.com/Expensify/App/issues/20404
- https://github.com/Expensify/App/issues/20522
- https://github.com/Expensify/App/issues/20611
- https://github.com/Expensify/App/issues/20929
- https://github.com/Expensify/App/issues/21560
- https://github.com/Expensify/App/issues/21616

Reply:
I don't see an alternative solution for now. Remounting is necessary to make this architecture work. Those are drawbacks of solutions that let us simplify the logic when the screen is not resized

Reply:
This is being actively worked on now by @adamgrzybowski and we will solve this by using css styles for the layout and we wont remount the navigators.